app/src/main/java/com/celzero/bravedns/service/ConnectionMonitor.kt

```diff
@@ -0,0 +1,240 @@
+     * else return the current active network.
+     */
+    fun getNetworkList(): LinkedHashSet<Network> {
+        val networks: LinkedHashSet<Network> = linkedSetOf()
+        if (persistentState.isAddAllNetworks) {
+            networks.addAll(currentNetworks)
+        } else {
+            currentActiveNetwork?.let { networks.add(it) }
+        }
+        return networks
+    }
+
+    private fun handleNetworkChange(isForceUpdate : Boolean) {
+        val message = constructMessage(if (persistentState.isAddAllNetworks) MSG_ADD_ALL_NETWORKS else MSG_ADD_ACTIVE_NETWORK, isForceUpdate)
+        serviceHandler?.removeMessages(MSG_ADD_ACTIVE_NETWORK, null)
+        serviceHandler?.removeMessages(MSG_ADD_ALL_NETWORKS, null)
+        serviceHandler?.sendMessage(message)
+    }
+
+    /**
+     * Constructs the message object for Network handler.
+     * Add the active network to the message object
+     * in case of setUnderlying network has only active networks.
+     */
+    private fun constructMessage(what: Int, isForceUpdate: Boolean): Message {
+        val message = Message.obtain()
+        message?.what = what
+        message?.obj = isForceUpdate
+        return message!!
+    }
+
+    // Handles the network messages from the callback from the connectivity manager
+    private class NetworkRequestHandler(looper: Looper?, val connectionMonitor: ConnectionMonitor) : Handler(looper!!) {
+        override fun handleMessage(msg: Message) {
+            // isForceUpdate - true if onUserPreferenceChanged is changes, the messages should be
+            // processed forcefully regardless of the current and new networks.
+            val isForceUpdate : Boolean = msg.obj as Boolean
+            when(msg.what) {
+                MSG_ADD_ACTIVE_NETWORK -> processActiveNetwork(isForceUpdate)
+                MSG_ADD_ALL_NETWORKS -> processAllNetworks(isForceUpdate)
+            }
+        }
+
+        /**
+         * Handles the active network.
+         * Get active network from connectivity-manager
+         * Set the vpn underlying network when the current active network is different from
+         * already assigned active network or if there is difference in the available network set
+         */
+        private fun processActiveNetwork(isForceUpdate : Boolean){
+            val newActiveNetwork: Network? = connectionMonitor.connectivityManager.activeNetwork
+            val isNewNetwork = connectionMonitor.currentActiveNetwork?.networkHandle != newActiveNetwork?.networkHandle
+            if (DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - ${connectionMonitor.connectivityManager.getNetworkInfo(newActiveNetwork)?.typeName.toString()}, $isNewNetwork")
+            connectionMonitor.currentActiveNetwork = newActiveNetwork
+            connectionMonitor.currentNetworks.clear()
+            if(!isNewNetwork && !isForceUpdate){
+                return
+            }
+            if (connectionMonitor.currentActiveNetwork != null) {
```

Comment:
the `if` below has a `<space>`, but the one above doesn't. Please follow a consistent styling of the code. It gets jarring to read otherwise.

Here's the Kotlin style-guide: https://developer.android.com/kotlin/style-guide This formatting should be a shortcut in Android Studio, so apply it to kotlin files?

Reply:
Added the space to the if statements. I think the formatting not happened for that code snippet. The auto format is set in the studio for Kotlin and Java files but some cases, it's not formatting.

Reply:
Editor already has the kotlin style formatter in-build in Android Studio and is set. Let me check the configuration again.

Reply:
Okay, thanks.